pype/plugins/global/publish/extract_review.py

```diff
@@ -13,22 +19,1064 @@ class ExtractReview(pyblish.api.InstancePlugin):
+
+        Prepares input and output arguments based on output definition and
+        input files.
+
+        Args:
+            output_def (dict): Currently processed output definition.
+            instance (Instance): Currently processed instance.
+            new_repre (dict): Reprensetation representing output of this
+                process.
+            temp_data (dict): Base data for successfull process.
+        """
+
+        # Get FFmpeg arguments from profile presets
+        out_def_ffmpeg_args = output_def.get("ffmpeg_args") or {}
+
+        ffmpeg_input_args = out_def_ffmpeg_args.get("input") or []
+        ffmpeg_output_args = out_def_ffmpeg_args.get("output") or []
+        ffmpeg_video_filters = out_def_ffmpeg_args.get("video_filters") or []
+        ffmpeg_audio_filters = out_def_ffmpeg_args.get("audio_filters") or []
+
+        # Add argument to override output file
+        ffmpeg_input_args.append("-y")
+
+        if temp_data["without_handles"]:
+            # NOTE used `-frames:v` instead of `-t` - should work the same way
+            duration_frames = (
+                temp_data["output_frame_end"]
+                - temp_data["output_frame_start"]
+                + 1
+            )
+            ffmpeg_output_args.append("-frames:v {}".format(duration_frames))
+
+        if temp_data["input_is_sequence"]:
+            # Set start frame
+            ffmpeg_input_args.append(
+                "-start_number {}".format(temp_data["output_frame_start"])
+            )
+
+            # TODO add fps mapping `{fps: fraction}` ?
+            # - e.g.: {
+            #     "25": "25/1",
+            #     "24": "24/1",
+            #     "23.976": "24000/1001"
+            # }
+            # Add framerate to input when input is sequence
+            ffmpeg_input_args.append(
+                "-framerate {}".format(temp_data["fps"])
+            )
+
+        elif temp_data["without_handles"]:
+            # QUESTION Shall we change this to use filter:
+            # `select="gte(n\,{handle_start}),setpts=PTS-STARTPTS`
+            # Pros:
+            #   1.) Python is not good at float operation
+            #   2.) FPS on instance may not be same as input's
+            start_sec = float(temp_data["handle_start"]) / temp_data["fps"]
+            ffmpeg_input_args.append("-ss {:0.2f}".format(start_sec))
+
+        full_input_path, full_output_path = self.input_output_paths(
+            new_repre, output_def, temp_data
+        )
+        ffmpeg_input_args.append("-i \"{}\"".format(full_input_path))
+
+        # Add audio arguments if there are any
+        audio_in_args, audio_filters, audio_out_args = self.audio_args(
+            instance, temp_data
+        )
+        ffmpeg_input_args.extend(audio_in_args)
+        ffmpeg_audio_filters.extend(audio_filters)
+        ffmpeg_output_args.extend(audio_out_args)
+
+        # QUESTION what if audio is shoter than video?
```

Comment:
I'm not sure if `-shortest` should be used here. I would expect to force video duration, maybe?

Reply:
I"d say that audio should never dictate the video length. Not in our usage. However we can certainly have shorter audio and it doesn't always start at 0. For example if animator has to extend a shot in Maya, but doesn't get new extended audio, the result should match the Maya timeline exactly.

Reply:
FIXED: Found another approach to this with setting `"-frames {video length}"` argument (hopefully works)